### PR TITLE
FIX _repr_html_ of Info add all channel types

### LINKED ISSUE #9707: _repr_html_ of Info missing many data channels
the _repr_html_ from Info class only shows a subset of channels eg misses iEEG and ECoG. The logic here https://github.com/mne-tools/mne-python/blob/main/mne/io/meas_info.py#L822 should be changed to get all data channels (even all channels) and show the counts.

### PR DESCRIPTION
**Pull request by vagechirkov**
Fixes #9707

### CODE REVIEW

**Comment by agramfort** [comment]
I've merged with main branch to attempt to fix CIs @vagechirkov

File: mne/io/meas_info.py [hunk 1]
```diff
@@ -815,37 +815,57 @@ def ch_names(self):
...
+        # good channels
+        channels = {}
+        ch_types = [channel_type(self, idx) for idx in range(len(self['chs']))]
+        ch_counts = Counter(ch_types)
+        for ch_type, count in ch_counts.items():
+            if ch_type == 'meg':
+                channels['mag'] = len(pick_types(self, meg='mag'))
+                channels['grad'] = len(pick_types(self, meg='grad'))
+            elif ch_type == 'eog':
+                pick_eog = pick_types(self, meg=False, eog=True)
```
Comment (larsoner) [inline mne/io/meas_info.py:833]: `meg=False` has been the default long enough I don't think we need it anymore. It was just there for the transition...
```suggestion
                pick_eog = pick_types(self, eog=True)
```

File: mne/io/meas_info.py [hunk 2]
```diff
@@ -815,37 +815,57 @@ def ch_names(self):
...
+                eog = ', '.join(
+                    np.array(self['ch_names'])[pick_eog])
+            elif ch_type == 'ecg':
+                pick_ecg = pick_types(self, meg=False, ecg=True)
```
Comment (larsoner) [inline mne/io/meas_info.py:837]:
```suggestion
                pick_ecg = pick_types(self, ecg=True)
```

File: mne/io/tests/test_meas_info.py [hunk 3]
```diff
@@ -830,6 +830,15 @@ def test_repr_html():
...
     assert 'Projections' not in info._repr_html_()
+    info['bads'] = []
+    assert 'None' in info._repr_html_()
+    info['bads'] = ['MEG 2443', 'EEG 053']
+    assert 'MEG 2443' in info._repr_html_()
+    assert 'EEG 053' in info._repr_html_()
+
+    html = info._repr_html_()
+    for ch in ['204 GRAD', '102 MAG', '9 STIM', '60 EEG', '1 EOG']:
+        assert ch in html
```
Comment (larsoner) [inline mne/io/tests/test_meas_info.py:841]: To make sure things work for new sensor types, can you please also add a smoke test like `assert 'Good channels' in raw.info._repr_html_()` to [mne/io/tests/test_raw.py:_test_raw_reader](https://github.com/mne-tools/mne-python/blob/2adf9a63218e932399710618a095305ae0bd5d0d/mne/io/tests/test_raw.py#L255)? This monolithic function is used by all `mne/io/*` reading functions, so it's a good test that it works.

**Review by larsoner — COMMENTED** [review]
Other than a minor suggestion to help with maintainability, LGTM!

**Review by agramfort — APPROVED** [review]
CI failure is unrelated.

@larsoner merge if happy

File: mne/io/meas_info.py [hunk 4]
```diff
@@ -815,37 +815,57 @@ def ch_names(self):
...
+                eog = ', '.join(
+                    np.array(self['ch_names'])[pick_eog])
+            elif ch_type == 'ecg':
+                pick_ecg = pick_types(self, ecg=True)
+                ecg = ', '.join(
+                    np.array(self['ch_names'])[pick_ecg])
+            channels[ch_type] = count
+
+        good_channels = ', '.join(
+            [f'{v} {k.upper()}' for k, v in channels.items()])
```
Comment (larsoner) [inline mne/io/meas_info.py:843]: On `main` I get:
Screenshot from 2021-09-15 13-58-04

On this PR it's a little bit less nice:

Screenshot from 2021-09-15 13-58-27

I prefer the longer / complete names, so some variant of this:
```suggestion
        titles = 
            [f'{v} {titles.get(k, k)}' for k, v in channels.items()])
```
I'll push a commit, can you see if you agree @vagechirkov ?

Comment (vagechirkov) [reply mne/io/meas_info.py]: I think it looks better with full names.

**Comment by larsoner** [comment]
Azure failure is an unrelated / spurious caching error after successful tests. @vagechirkov can you add a `latest.inc` entry for this fix? Then I'll merge!

**Comment by larsoner** [comment]
Pushed a commit to move the change to below those of new contributors, then will merge once CIs come back happy. Thanks in advance @vagechirkov !

**Comment by agramfort** [comment]
thx @vagechirkov and @larsoner !

**Pull request merged**